src/VisualStudio/Core/Def/Implementation/ProjectSystem/AbstractProject.cs

```diff
@@ -1015,9 +1016,12 @@ protected virtual void OnDocumentRemoved(string filePath)
         {
         }
 
-        protected virtual void UpdateAnalyzerRules()
-        {
-        }
+        /// <summary>
+        /// Implemented by derived types to provide a way for <see cref="AbstractProject"/> to indicate that options will need to be refreshed.
+        /// It is expected that derived types will read in shared option state stored in this class, create new Compilation and Parse options,
+        /// and call <see cref="SetOptions"/> in response.
+        /// </summary>
+        protected abstract void UpdateOptions();
```

Comment:
The current model is that AbstractProject is well, abstract. We could do that but that'd be a larger refactoring that we shouldn't try doing here.